app/models/sitemap.rb

```diff
@@ -0,0 +1,110 @@
+require "builder"
+
+class Sitemap
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  include Sitemap::RobotsGeneratable
+
+  ERROR_PAGES = %w[400.html 404.html 406-unsupported-browser.html 422.html 500.html].freeze
+
+  attribute :base_url, :string
+  attribute :build_dir, :string
+  attribute :generate_robots, :boolean, default: false
+
+  validates :base_url, presence: true, format: { with: URI.regexp(%w[http https]) }
+  validates :build_dir, presence: true
+
+  def generate!
+    write_sitemap!
+    write_compressed_sitemap!
+
+    handle_robots_generation!
+
+    total_pages
+  end
+
+  private
+
+  def write_sitemap!
+    File.write(sitemap_path, xml_sitemap)
+
+    Rails.logger.info("Generated sitemap.xml with #{total_pages} URLs")
+  end
+
+  def write_compressed_sitemap!
+    Zlib::GzipWriter.open(sitemap_path(compressed: true)) do |gzip|
+      gzip.write(minified_xml_sitemap)
+    end
+
+    Rails.logger.info("Generated sitemap.xml.gz (compressed)")
+  end
+
+  # XML generation
+  def xml_sitemap
+    @xml_sitemap ||= build_xml_sitemap
+  end
+
+  def minified_xml_sitemap
```

Comment:
This minification approach is too aggressive and may break XML. Consider using a proper XML minifier or keeping the formatted XML since the gzipped version will compress whitespace effectively anyway.